frontends/p4/simplifyDefUse.cpp

```diff
@@ -545,6 +551,13 @@ class RemoveUnused : public Transform {
         }
         return statement;
     }
+    const IR::Node* preorder(IR::Function* func) override {
```

Comment:
do you have an example where this is important?

Reply:
I extended the virtual.p4 test case to include the problematic case -- an abstract method with an inout arg where the implementation assigns to the arg (could also happen with an out arg) -- not skipping the method here would remove the assignment (leaving the method empty).

Reply:
With the rules I proposed this would be correct, because no one is calling this method.
We need to know who is calling the method to analyze it.
If we don't know the method is either asynchronous or dead.